servers/physics_2d/space_2d_sw.cpp

```diff
@@ -671,21 +680,15 @@ int Space2DSW::test_body_ray_separation(Body2DSW *p_body, const Transform2D &p_t
 				break;
 			}
 
+			recover_motion /= recover_count;
+
 			body_transform.elements[2] += recover_motion;
 			body_aabb.position += recover_motion;
 
 			recover_attempts--;
 		} while (recover_attempts);
 	}
 
-	//optimize results (remove non colliding)
```

Comment:
This part is not needed anymore, since we're considering hits at depth 0 as a collision now, in order to stabilize floor tests.

As a side note, this block was not working correctly, as it would skip one element after each swap.
Proper implementation would have been:
```
for (int i = 0; i < rays_found; i++) {
	if (r_results[i].collision_depth == 0) {
		rays_found--;
		SWAP(r_results[i], r_results[rays_found]);
		i--; // because we need to check the element we just moved
	}
}
```